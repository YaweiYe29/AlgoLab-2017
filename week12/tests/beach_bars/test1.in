5
780
50 -739 -252 814 313 -875 856 428 541 682 564 -530 -551 -114 -238 -695 417 -404 -730 626 -263 -350 -225 905 -306 456 -428 1000 602 981 -852 381 998 -297 578 -490 -705 651 9 532 993 13 772 -793 563 -207 -765 105 559 -944 -158 -401 784 220 152 68 -692 -680 824 -120 167 606 938 667 -689 -498 164 609 425 -931 -299 577 821 -136 677 -41 -213 -219 340 -675 -958 788 771 115 210 -587 929 -427 -370 -360 -540 224 -459 100 -760 28 825 -859 350 851 -959 890 248 -610 -986 -138 206 -898 -101 -588 -497 785 -571 -905 -881 675 -579 -694 -997 -200 812 -953 214 211 -348 711 580 -776 833 849 555 222 82 638 -865 157 -409 150 45 842 332 -323 635 -361 -965 -168 -844 237 593 551 -780 312 284 -210 159 -924 692 781 470 -886 -508 176 489 -867 829 32 23 351 705 -273 286 -479 -595 -352 207 -254 -314 245 689 -96 -397 162 684 589 479 -569 787 827 226 566 -790 985 384 -272 287 277 70 267 -748 472 -87 344 -438 796 -296 333 -632 724 -251 -895 461 850 663 -618 834 37 673 -478 -132 -722 634 -450 -624 137 180 26 -434 -984 817 -690 -121 438 -918 -496 185 -374 982 953 156 -640 -341 786 -757 526 -932 -135 -787 192 989 -280 -309 250 601 316 -243 -461 970 -386 14 811 -827 758 -163 -758 575 -781 -742 -13 215 -658 698 -385 174 971 -993 363 -196 -88 -885 490 -866 770 -586 717 418 -938 645 854 636 888 -468 91 -973 -855 501 -521 515 -841 258 -301 680 -580 776 -431 65 538 -21 766 -145 414 325 -265 -212 -441 118 -706 -149 410 400 -956 -771 585 74 -485 -948 408 -169 -772 -854 -329 742 672 55 -504 745 -387 172 -317 642 -714 288 -439 917 120 583 -773 -218 47 844 -46 -843 -29 567 136 933 -423 371 -985 -534 -511 48 845 -61 -858 -957 -589 767 161 -391 168 44 -515 378 -115 444 -912 111 -448 557 -733 160 374 -185 -919 -258 96 893 -463 -487 -600 299 -260 853 56 -417 -794 -482 77 474 -419 759 -597 303 -222 -594 762 -528 -556 594 -762 -817 536 -23 -86 -883 473 -437 919 -768 270 -623 -443 913 116 -223 336 -92 -952 404 -35 376 -183 -980 71 -91 468 -822 -250 309 306 -647 246 -343 -599 -10 549 615 -741 -581 273 -286 607 429 -464 983 186 736 -620 182 979 134 931 600 -756 -616 -642 797 0 -327 183 -429 -892 -95 464 527 -32 861 271 830 -526 354 -34 -831 525 988 -368 972 175 -342 113 -806 87 547 -12 -410 149 612 -744 -740 616 -752 604 737 -553 -906 450 -769 984 -372 -613 280 83 621 -735 -382 114 679 -172 -635 721 -54 -851 836 146 -416 940 831 -922 434 -964 -902 -105 613 -743 647 -945 -148 411 808 991 -682 -285 459 522 -37 -621 735 -555 367 -192 -989 364 -195 -592 455 -835 658 -235 656 -700 104 901 -452 107 -961 -164 -627 729 750 -606 506 -850 -916 440 -968 244 -315 -943 413 802 5 -891 -94 -157 402 -102 420 -73 486 -870 483 -76 -663 -266 -274 -862 -65 297 -262 -307 252 -890 -93 -887 353 416 -143 -737 553 -261 -637 -303 256 22 819 -537 -934 -166 330 -516 -979 377 -913 -813 630 -270 289 -467 92 990 -97 -894 -563 793 357 -999 123 520 -36 523 -764 129 -201 295 -798 -67 492 641 -715 -868 -871 -74 625 -665 181 978 -552 760 -596 84 -475 818 21 809 948 -408 548 -808 -11 -52 710 -646 247 -179 -976 -290 -197 362 -518 41 -14 -811 942 -414 482 85 -788 568 171 448 -111 -908 -177 382 579 385 -63 213 -242 -462 97 460 110 -783 67 -492 276 -617 629 924 846 -510 930 -357 139 355 -204 -795 -745 -558 1 963 166 -393 500 900 903 106 -133 586 -118 441 707 -649 -837 -40 519 -224 748 -608 690 -666 290 -269 -951 -910 46 843 -513 -903 778 855 -501 -631 725 394 -962 -612 281 147 -412 163 960 763 -50 509 659 941 144 135 932 -15 544 -728 112 909 -334 -717 726 329
817
249 713 -683 498 827 -429 -729 951 -738 737 2 -319 857 -830 92 -162 760 459 86 -836 -350 572 933 296 -626 -612 -444 478 -597 -857 616 176 309 60 -862 -269 -916 -279 643 174 -748 -151 272 960 274 -902 -55 515 119 -440 -605 777 164 -758 -931 -926 345 -214 -704 218 -273 -636 286 -391 -603 -330 592 -335 -356 566 -362 192 -730 491 798 575 -984 -379 577 -717 974 -211 -954 -248 674 648 -274 -229 885 -471 -403 -4 918 963 923 -903 19 816 -322 600 605 -441 -43 172 -750 -604 324 -598 98 -824 217 846 412 -510 761 -161 278 -644 -570 127 -795 -399 507 -415 -514 700 -222 -237 685 -671 72 -493 -856 66 624 -298 -753 -173 749 -739 183 80 -842 -45 -25 897 -825 -142 964 832 -524 202 5 -917 -427 495 275 11 791 -131 852 -70 928 326 894 -462 234 -688 -749 -761 -798 124 -651 271 634 -288 -802 -167 508 -414 145 678 -244 -654 268 -255 667 679 -243 -606 316 -600 322 -608 -184 397 -959 545 -377 800 -122 -270 652 -735 -448 474 -733 -241 671 -18 -512 410 -614 952 -312 -231 691 427 -495 -109 -580 342 815 472 -992 747 900 -213 367 -192 720 -202 157 937 503 -419 557 -367 -722 -660 178 -744 -584 304 -618 223 122 421 -935 -726 -627 295 -264 230 593 -329 -508 414 -942 -240 698 849 52 -205 717 -787 -280 893 -826 96 -366 556 -344 -554 368 -988 -191 -835 -38 -208 -201 939 -21 104 -818 78 -602 34 -742 615 787 -155 618 -304 -449 -812 110 -982 -165 -67 855 -126 33 830 396 -526 701 -221 -896 26 -754 669 321 -601 783 -573 294 -628 475 -447 909 212 -936 54 -266 656 260 -662 568 -788 379 -543 16 -363 993 -59 500 147 944 -152 10 -912 423 -499 -773 985 -553 369 -987 676 -246 50 -872 847 -75 484 -669 687 914 -8 -805 -272 -684 328 544 -378 181 -219 -509 413 -943 -146 -813 109 969 -961 -164 -457 -820 102 -84 790 -132 -930 -496 426 -133 -572 95 -464 442 94 -465 -228 -591 331 -529 765 -661 -436 486 -588 334 697 -225 854 -502 57 -57 865 431 -491 -854 68 -767 -116 806 -550 740 -619 1000 574 -782 -451 905 150 -772 -168 391 799 -557 245 -101 -898 -672 250 613 -309 46 -79 843 -513 -476 446 -113 -910 477 840 -82 -953 456 -900 384 -538 21 -281 -305 -668 254 148 945 -276 283 -968 -171 469 -401 158 81 -454 902 -817 105 47 -875 874 -482 -845 77 -634 288 -655 267 89 -470 886 527 -395 961 -973 383 235 632 -486 -849 73 -358 564 -642 116 -9 913 343 -216 -566 356 -203 204 998 400 -93 -890 0 797 -972 -932 -892 362 -197 -121 441 804 -118 7 -915 243 -679 -708 214 130 621 -301 370 -552 -978 378 -838 55 -583 773 -183 -980 -546 376 -299 726 692 -230 329 931 449 -473 -62 497 -86 836 539 936 371 -188 -320 306 128 -794 562 -907 512 -47 -200 359 756 241 -652 -766 -406 950 -6 553 -586 -549 373 -983 252 -670 -112 13 -909 381 844 -599 723 -199 -996 -957 -308 406 -950 803 -144 781 -141 -938 858 -498 61 -140 782 -871 51 -426 99 -823 -632 259 -925 -128 -967 -170 355 -567 82 79 876 -877 -80 479 -757 -615 162 -179 -23 536 -102 631 660 -262 690 -666 227 673 -249 -311 248 -204 718 -638 946 -455 -421 501 588 37 -278 -294 -143 -506 416 25 822 28 570 -686 236 -323 922 -810 546 -74 85 882 -474 419 845 -77 688 735 -621 -283 213 -829 -32 220 143 934 -819 537 -664 -267 -257 665 908 511 -537 115 -1 492 -398 -620 736 -99 -966 -488 71 868 216 -343 785 237 828 976 -743 179 -714 732 -883 -89 -483 -977 710 663 107 -452 -302 988 -640 -256 666 -690 997 209 -960 547 -809 -12 -409 684 -172 750 769 972 975 269 -174 -56 -519 837 70 -489 -61 401 -521 835 38 432 461 888 488 -373 186 983 -460 62 -860 -63 -69 -866 968 -582 774 -83 -921 766 -933 -177 382 480 510 -239 920 986 991 -296 263 -995 -207 -349 138 926 528 -762 -595 48 -831 -523 -147 -189 -986 627 850 53 -969 387 -694 996 199 -154 -351 -676 411 170 -408 548 -11 -52 841 -796 -993
804
-599 930 -478 -271 983 812 -110 142 -539 480 -341 219 639 -184 738 -57 865 -256 666 -297 625 -731 -754 168 -242 -727 943 146 866 -304 -655 -40 882 284 -638 -598 369 499 -337 -592 330 -755 167 313 -609 -121 801 904 155 -767 -896 25 -897 623 -299 260 293 -629 439 -483 767 -292 527 748 628 -294 412 -510 -967 320 -602 539 902 -20 -53 72 -850 928 -791 131 802 -130 792 -660 -226 696 -476 915 -7 132 -790 -650 272 -552 115 -807 699 382 259 -663 157 -765 -298 261 -710 -165 282 -866 -903 19 -207 715 -125 -573 783 -571 425 -61 861 762 -726 338 -584 569 -231 119 482 -440 -161 -524 398 -406 -607 -62 868 -54 685 -864 -67 502 879 -43 641 -575 -634 -169 529 -77 -511 486 256 166 325 -741 -738 193 -729 136 -137 -934 -400 159 337 -468 -771 601 -321 161 712 -831 -34 525 -968 751 -171 -211 711 -506 416 -823 99 462 -460 750 316 -606 883 319 -603 133 -789 -995 -198 586 -336 533 -389 170 217 -705 -356 857 60 253 -669 -517 283 476 -536 386 -86 836 -633 -218 704 -98 -461 461 -96 387 190 -732 -444 478 -878 544 176 360 -996 496 -426 -566 -259 321 957 94 -693 -437 859 -860 62 -572 350 -209 -830 -33 -396 526 -175 -990 923 -331 -838 611 -311 -930 -898 24 302 -620 -186 736 705 938 -418 -113 332 791 954 540 -382 974 186 -736 -243 -50 700 -937 -140 -793 129 0 931 -425 503 800 3 -541 541 -381 975 273 -649 -512 844 -547 -225 697 -306 616 726 -196 363 8 -914 -117 582 22 853 -877 45 674 -248 574 198 -724 -616 -471 88 -37 885 795 -561 -386 -975 -624 840 -82 678 -568 -432 -795 127 -615 741 -818 104 467 -455 -21 901 798 555 717 -427 -971 385 -330 -235 276 281 165 -757 407 -515 -11 -808 373 -983 -679 243 -316 258 -301 406 -516 668 234 -688 -956 -522 400 -595 -894 28 -97 825 -350 -713 209 847 -387 138 -784 -293 266 -129 793 359 -563 -960 -9 913 -806 116 560 -362 197 189 552 -370 -630 807 561 -361 -365 557 -2 716 -317 -680 242 66 418 -504 510 -285 987 -839 517 -42 -502 854 -865 57 264 982 -374 -816 -201 63 -496 794 -912 -481 441 -118 909 -447 834 -885 37 -944 -980 376 -282 640 -849 73 -486 777 -832 90 887 164 -758 598 998 201 -997 -977 -543 379 -871 51 -52 65 899 -23 -623 299 -260 -47 875 378 -544 15 -84 -384 538 -653 329 -593 684 -672 528 -31 -861 -64 -272 287 -401 521 884 -38 -835 21 -901 455 -220 339 784 -274 -373 -802 120 917 -289 633 199 -723 -262 297 519 59 -187 -984 341 -722 82 -964 326 -224 698 210 -454 105 -109 813 -906 16 413 -146 -305 617 511 595 -577 89 -833 941 -415 -44 878 -841 81 -148 774 77 780 227 -600 322 -772 584 -889 596 -93 829 32 604 270 207 139 -102 -380 179 960 -689 233 -312 610 676 -869 487 903 -453 40 -519 -85 837 -916 440 263 -296 -325 597 -221 720 286 -636 -702 654 918 -769 900 -22 103 -819 71 -488 -329 230 656 -538 355 310 501 -421 935 -355 675 -681 -359 997 192 -216 343 -950 -153 796 -126 433 -923 -926 -526 -456 730 -192 304 -626 231 -328 565 562 962 959 961 -745 277 808 423 -422 -785 137 -588 334 -162 335 771 -505 109 -16 906 972 575 -781 619 -737 -26 897 -459 100 -484 75 932 498 -424 -27 437 -344 -770 152 543 940 -103 -404 952 -17 876 -414 -368 988 357 -886 -197 -991 30 827 493 -66 427 -466 -909 481 -441 657 -265 294 -798 955 158 251 -71 488 -627 729 395 -164 -472 -751 171 968 733 -189 -586 254 -792 -168 128 925 -947 409 803 653 -637 -766 524 247 714 -208 -728 709 -213 38 -458 -855 886 545 -570 -686 670 183 -376 452 849 810 766 35 -921 -371 125 737 -219 340 -891 652 -270 424 -932 -978 -647 -981 268 665 -578 -569 -704 649 -273 -614 864 -29 667 745 -707 226 -333 -796 -279 -342 27 789 96 556 -3 -685 635 -888 799 -976 -985 -718 638 -343 947 631 669 -621 622 710 881 484 -457
505
745 -635 -272 -295 858 886 -837 -772 -448 105 -937 -234 523 427 980 -864 -780 719 898 -836 114 834 -735 -768 15 -564 345 700 970 829 -91 331 -88 130 -447 -912 138 -753 747 553 350 124 67 -994 263 740 1000 241 332 -978 671 -182 510 -769 390 727 -629 -259 5 246 -599 -68 -361 242 -205 -992 617 978 382 -725 -546 -726 -779 -918 663 228 -331 -789 -809 -608 748 -420 936 139 185 -374 -93 737 -619 -485 402 -954 -733 623 -681 675 509 -50 446 132 174 -778 -422 416 -940 -250 -793 563 -106 -903 53 -506 527 947 150 798 -558 329 30 827 698 -658 -467 92 941 271 324 763 -593 -271 200 997 525 -831 -703 599 973 -383 -20 539 -895 582 -774 169 -920 -123 -484 607 -749 -343 216 -476 880 -989 326 -233 255 -304 712 -644 -710 646 249 455 -901 -246 -152 -949 407 -373 186 -729 730 -957 725 -631 685 -671 -737 619 -180 -977 606 238 820 -70 -867 -265 470 -868 -936 42 -851 -369 -473 817 -119 257 654 -19 540 -176 383 -510 -930 -56 677 280 -653 637 -319 240 -999 -268 291 -1 -795 717 -639 651 -528 -334 282 513 -843 -507 -375 -755 932 650 170 -389 504 95 68 -491 146 -813 543 551 -8 -805 681 -687 134 931 -581 775 -915 -118 433 -348 211 236 -657 -688 -877 479 -604 -64 2 -155 738 319 -971 -77 -822 -393 412 556 -800 -326 -723 772 375 -708 979 -189 -983 -99 397 -496 794 797 0 -329 230 293 -266 199 -694 -138 -932 358 -341 949 870 -23 -889 467 -823 814 -832 -35 -523 -456 437 40 708 955 624 -732 -782 855 414 14 811 -826 -607 286 -665 691 297 -158 998 824 -863 -577 3 -556 -953 -159 337 -419 140 -165 -962 478 -716 -922 -125 625 -146 413 926 129 526 825 -796 -402 -597 759 -166 393 869 -802 91 -601 292 -667 689 66 570 -659 -652 -255 359 822 -534 -441 118 956 -400 159 -337 -217 676 882 85 910 -777 80 77 874 480 -960 -669 701 -505 851 -42 -633 195 -698 258 -301 989 795 398 81 -478 415 -875 272 -224 766 569 372 425 -127 -924 -128 327 -842 -251 7 465 528 -31 -722 965 -278 -672 848 451 -595 -902 141 -818 -21 538 96 890 -617 229 -664 626 -661 232 -237 815 -872 818 21 106 -62 497 894 -339 707 -477 942 516 -43 -503 853 56 -365 128 591 -765 125 922 -715 334 -512 -909 838 -625 268 736 548 154 -514 842 445 664 -382 952 977 -116 -513 742 261 -785 -226 270 -623 4 -642 -933 -457 -438 -576 -173 -970 449 831 208 -241 823 -714 573 -85 961
971
899 -862 479 14 -851 9 -738 -486 -151 -122 464 -930 729 686 -116 -82 -909 620 364 -558 905 533 967 927 11 -648 -345 276 596 -177 679 914 -376 -69 138 -353 896 -360 -395 89 18 628 214 -38 982 827 112 346 956 249 -543 272 -986 144 -521 -779 -942 -625 297 -46 -473 366 -536 386 -592 587 -19 903 -939 537 -477 381 -541 -25 702 -112 -587 247 930 -234 487 -842 595 240 593 -664 -988 29 -459 -551 371 952 198 250 779 -143 584 443 501 -277 20 394 -291 -129 -462 416 -506 49 -712 210 -267 655 375 -958 -951 580 -342 821 -101 -80 -955 -591 331 154 -768 574 473 -449 -586 336 517 665 -257 -667 255 -659 -559 363 -478 -303 -164 -740 790 -132 -84 838 -763 159 553 -369 949 782 -453 279 -188 -287 715 -207 -662 260 -610 312 -409 695 -595 332 25 -47 295 613 -743 -271 651 -83 839 37 -885 -62 860 -815 626 195 -727 921 -1 -176 -533 -896 26 -781 141 504 -418 670 -251 -476 491 -431 894 -979 161 -254 -745 17 -905 -351 571 160 957 246 -676 680 -679 243 -12 910 216 -706 -272 650 -866 96 -826 530 733 299 -623 986 241 342 -580 -538 -697 486 -952 -155 404 -194 728 993 329 550 -372 872 -221 -567 355 713 -505 417 -310 612 570 -352 -422 500 833 -89 -452 470 226 673 -248 397 -925 -649 273 589 -707 592 -330 -501 421 -937 578 -344 -455 -21 901 -130 792 -337 585 -860 62 -63 859 -284 850 687 -235 58 -864 826 -96 478 392 -835 87 168 966 282 -640 -605 317 357 -565 -150 -33 889 804 -118 7 -915 -916 -922 -990 -606 292 -630 -106 19 622 -300 -432 333 -589 -639 283 -888 34 709 -213 -974 -469 -439 483 169 -753 885 -37 522 485 -437 -819 -275 647 552 617 -305 46 -876 -653 269 932 135 -283 639 301 -750 606 730 738 -618 -413 509 542 -814 407 -515 -658 -211 86 -836 883 -39 -339 583 -892 -256 666 -269 653 219 -703 -620 -156 766 731 900 466 -456 -290 632 -346 776 -146 985 -723 917 -5 120 -802 -368 554 130 720 -340 -615 307 253 -692 230 -401 521 912 -996 -997 -328 231 911 114 -445 281 -278 239 602 -320 -208 714 -481 131 563 94 -828 -295 961 164 93 456 -466 -32 890 -509 689 403 472 -450 85 -837 519 -734 770 -152 -949 -772 916 832 -524 700 -222 337 459 -100 753 -169 -626 296 -931 -134 -960 227 -232 -394 962 -285 637 203 -719 717 -205 354 22 -537 -919 954 -423 340 -582 146 -776 67 430 -492 322 -600 -311 -674 248 -843 508 -414 -777 145 -875 -78 481 -322 614 -308 325 -816 287 -635 -86 869 -53 506 -145 777 343 -579 -882 40 -85 837 -970 845 -511 -375 547 380 743 -179 -54 -417 505 105 468 -454 902 140 579 -343 -991 -557 365 -396 526 4 367 -555 489 581 298 -624 185 -737 -307 615 920 945 -411 259 -663 693 180 -742 -596 760 -671 -144 -941 -313 820 -102 280 643 -279 361 -561 795 54 -868 488 -468 -831 91 994 197 344 557 656 -700 -796 560 310 -44 81 -841 909 913 -9 -406 -281 641 -715 -282 640 461 824 -98 -895 408 -485 118 915 55 -504 -355 567 -789 356 -566 -51 111 867 -55 -849 -147 775 372 -187 290 -206 -35 -832 -398 524 873 -49 -846 38 -66 856 422 -500 997 -577 -588 618 -502 -499 423 -899 23 191 61 -861 -64 -123 799 -160 306 -616 -182 740 328 -594 -934 -834 88 -471 -912 10 -729 -603 319 -286 636 -654 -216 -382 -183 739 -617 347 -178 744 789 -378 851 284 694 565 886 -36 -65 857 -824 98 -475 156 953 -231 -694 662 -657 265 -157 -520 402 -91 -489 433 -923 -126 224 669 922 888 -34 525 870 661 -195 727 -629 -512 -115 -460 99 893 -30 -827 -393 -790 100 -159 664 -655 267 -48 -474 448 -637 285 -240 682 624 -298 975 -487 72 -780 -746 176 -383 16 -765 591 -652 866 -56 -853 834 303 66 863 -390 532 -756 642 -735 -163 759 725 -197 -560 362 428 -928 70 201 992 -301 192 989 -818 -384 538 935 -879 -58 438 -94 828 751 -571 351 242 -680 676 -553 369 610 710 -980 313 -977 940 -713 209 -350 175 -747 -16 -379 543 -43 812 -907 15 -881 -944 -510 412 685 -237 -634 -914 339 -583 773 373 39 -920 236 -60 -726 -292 630 36 -783 139 936 452 -838 605 -714 208 996 151 -408 948 754 882 843 68 -854 502 -57 -917 -120 439 45 117 -805 561 -361 -218 341 -552 -749 278 -15 -812 -315 244 703 -214 708 -732 977 414 748 -962 -670 964 -755 167 -250 672 -325 -722 783 -507 809 -113 -910 -863 -757 -760 315 -607 286 359 -563 787 -569 784 -572 -969 -175 -633 496 183 980 663 -693 -890 -373 -270 652 -604 318 -241 -545 811 814 999 211 -926 -529 -124 877 943 -24 535 28 204 -718 972 -226 -198 -548 -945 24 465 -391 531 -825 277 -975 424 -297 149 946 -689 601 -736 781 -265 -613 -752 604 -289 -950 -647 -978 755
