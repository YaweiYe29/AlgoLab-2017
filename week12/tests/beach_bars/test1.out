92 100
85 86
103 100
-212
90 100
33 38 39 42 43 46 187 188 862
67 100
-723 -697 -696 -695 -693
111 100
275
